### DISCUSSION TITLE
Lessons learned connecting SPI for CC1101 on STM32 and ESP8266 NodeMCU

### DISCUSSION THREAD

**Discussion by sewnbacktogetherwrong** [jgromes/RadioLib#804] [Q&A]
Hi all not a question but sharing some stuff I figured out as a newbie after a lot of frustration.

First I connected the CC1101 to my STM32F108 Blue Pill.
The examples show the following for SX1278 board:
// SX1278 has the following connections:
// NSS pin:   10
// DIO0 pin:  2
// RESET pin: 9
// DIO1 pin:  3
//SX1278 radio = new Module(10, 2, 9, 3);

For the STM32 Blue Pill, you need to use MOSI1, MISO1, SCK1, and then choose other data pins to be NSS/CS and GD0. You also need to use the STM32 pin names. Here was my working configuration:
// CC1101 has the following connections:
// CS pin:    PA4
// GDO0 pin:  PA2
// RST pin:   unused
// GDO2 pin:  3 (optional)
// For STM32
 CC1101 radio = new Module(PA4, PA2, RADIOLIB_NC, 3);

For EDP8266 NodeMCU while you think you should use the standard MOSI/MISO/SCK pins, RadioLib apparently expects to use the HMOSI, HMISO, HSCK pins on the other side of the board. Again you need to choose two other pins for CS and GDO. Don't choose the RX/TX pins because those are used for the serial connection back to the PC. Here was my working ESP8266 NodeMCU configuration:
CC1101 radio = new Module (16,5,RADIOLIB_NC,4);

**Comment by jgromes** [comment]
> For the STM32 Blue Pill, you need to use MOSI1, MISO1, SCK1

Not really, you need to use whichever pins your platform/Arduino defines as `SPI`. That's not really up to RadioLib, unless you specify otherwise, it will use the default SPI instance.

> For EDP8266 NodeMCU while you think you should use the standard MOSI/MISO/SCK pins, RadioLib apparently expects to use the HMOSI, HMISO, HSCK pins

Again, same thing - ESP8266 in your selected configuration (NodeMCU) defines that HSPI should be used for the default `SPI` instance. RadioLib doesn't touch that configuration unless you explicitly tell it to by using a non-default Module constructor.

I would suggest taking a look at the [Basics Wiki page](https://github.com/jgromes/RadioLib/wiki/Basics).
